docs/software/koji-workflow.md

```diff
@@ -47,53 +47,137 @@ The results of a buildArch task. Their metadata are recorded in the Koji databas
+    This will allow you to make some changes using the [Koji web interface](https://koji.opensciencegrid.org/koji).
 
--   Add this directory to your `$PATH`
--   Run
+1.  Set up the OSG Koji config
 
         :::console
-        [you@host]$ osg-koji setup
+        you@moria$ osg-koji setup
 
-    to set up the koji configuration and certificates in `~/.osg-koji`
+    Answer "yes" to all questions.
 
-### Obtaining a login
 
-You will be using your grid certificate to log in. Email a Koji admin the DN of your certificate, and we will set up a Koji account with the appropriate permissions.
+### Setting up on your own host
 
-If you are switching certificate providers, you will need to email a Koji admin with your new DN. You will also need to clear your browser cookies and cache for `https://koji.opensciencegrid.org` before trying to use the Koji web interface again. If your CN has changed, you will not be able to use your old certificate.
+This requires an Enterprise Linux 6 or 7 host.
 
-Current Koji admins are Mat Selmeci and Carl Edquist.
+1.  Install the [OSG YUM repositories](https://opensciencegrid.org/docs/common/yum/)
 
-### Configuring certificate authentication
+1.  If using OSG 3.5 or newer, enable the `devops` repository.
 
-You must also configure certificate authentication for the command-line tools on your build host:
+1.  Install osg-build and its dependencies:
 
--   Run
+        :::console
+        you@host$ sudo yum install osg-build
+
+1.  Install a program for getting grid certificates
 
         :::console
-        [you@host]$ osg-koji setup
+        you@host$ sudo yum install globus-proxy-utils
 
-    to set up the appropriate configuration and certificates in `~/.osg-koji`
+    !!! note
+        If you already have `voms-clients-cpp` or `voms-clients-java` installed,
+        you can use `voms-proxy-init -rfc` instead of `grid-proxy-init`,
+        and don't need to install `globus-proxy-utils`.
 
-After this, you will also be able to run koji commands manually by using the `osg-koji` wrapper script. You might need to rerun `osg-koji setup` if you renew or change your cert.
+1.  (Optional) If you want to do mock builds (these are local builds in a chroot), add yourself to the `mock` user group:
+
+        :::console
+        you@host$ sudo usermod -a -G mock $USER
+
+1.  Copy your user certificate and key into `$HOME/.globus/usercert.pem` and `$HOME/.globus/userkey.pem`.
+    Make sure `userkey.pem` is only readable by yourself.
+
+    !!! note
+        If you are using a certificate from SAML or Kerberos credentials, such as with `cigetcert` or `kx509`,
+        skip this step.
+
+1.  (Optional) Load your certificate into your browser.
+    This will allow you to make some changes using the [Koji web interface](https://koji.opensciencegrid.org/koji).
+
+1.  Set up the OSG Koji config
+
+        :::console
+        you@moria$ osg-koji setup
+
+    Answer "yes" to all questions.
+
+
+Logging In to Koji
```

Comment:
I think it feels a bit weird to call it "logging in to koji", if all you are doing is making a grid proxy (and that step doesn't communicate with koji at all).  The verify step isn't really logging in either, it's just testing your access.

I'm not sure what to call it instead, but something about it sounds slightly off to me.

Reply:
I changed it to "authenticating with koji", does that read better?

Reply:
Sure, it's an improvement.  And i mean however you think about it is fine.  But yeah, to me, "logging in to koji" is what i do on the koji web interface when i hit "login".

The way i read it though, the point of this section is more like "establish credentials for koji". You do in fact "authenticate" with koji if you run the test step at the end, but that's just a confirmation step to verify that the main point (establishing credentials) was successful.

This actually first caught my attention [later](https://github.com/opensciencegrid/technology/pull/748/files#diff-f091234d2b0a018d3547e587bbe164a1R284) in the changes for this PR, where it says:

> Before using `osg-promote`, **log in to Koji as above**

Because it's not like you need to "log in to koji" first (or even "authenticate with koji" first), you just need to make sure you have established your credentials that will be used for koji.

And a further small detail about my wording: while it might very slightly more concise to say "establish koji credentials" (rather than "establish credentials for koji"), i would find that slightly misleading, since your grid credentials are not specific to koji; koji just happens to be one place where you use them.

Reply:
"establish credentials for koji" sounds like fine phrasing.

I'm going to turn your last comment around: your Koji credentials are not specifically grid credentials, since Koji has its own CA that can make them. Right now it's only been used for daemons and automated users, and all current humans users have credentials that also work on the grid (well, don't know about the Fermi certs you get when you run `kx509`), but me handing out certs is kinda a fallback plan in case (once?) OSG moves off of GSI.